design/cli-install-changes.md

```diff
@@ -0,0 +1,286 @@
+# Proposal for a more intuitive CLI to install and configure Velero
+
+Currently, the Velero CLI tool has a `install` command that configures numerous major and minor aspects of Velero. As a result, the combined set of flags for this `install` command makes it hard to intuit and reason about the different Velero components. This document proposes changes to improve the UX for installation and configuration in a way that would make it easier for the user to discover what needs to be configured by looking at what is available in the CLI rather then having to rely heavily on our documentation for the usage. At the same time, it is expected that the documentation update to reflect these changes will also make the documentation flow easier to follow.
+
+This proposal prioritizes discoverability and self-documentation over minimalizing length or number of commands and flags.
+
+## Goals
+
+- Split flags currently under the `velero install` command into multiple commands, and group flags under commands in a way that allows a good level of discovery and self-documentation
+- Maintain compatibility with gitops practices (i.e. ability to generate a full set of yaml for install that can be stored in source control)
+- Have a clear path for deprecating commands
+
+## Non Goals
+
+- Introduce new CLI features 
+- Propose changes to the CLI that go beyond the functionality of install and configure
+- Optimize for shorter length or number of commands/flags
+
+## Background
+
+This document proposes users could benefit from a more intuitive and self-documenting CLI setup as compared to our existing CLI UX. Ultimately, it is proposed that a recipe-style CLI flow for installation, configuration and use would greatly contribute to this purpose.
+
+Also, the `install` command currently can be reused to update Velero configurations, a behavior more appropriate for a command named `config`.
```

Comment:
Do you want to remove the reference to `config` here given the comment that `config` is being dropped?

Reply:
Thanks, will do.